ctapipe/io/dl1writer.py

```diff
@@ -468,6 +487,34 @@ def fill_from_simtel(
                         containers=hist_container,
                     )
 
+    def _write_muon_events(self, writer: TableWriter, event: ArrayEventContainer):
+        """
+        add muon entries to the event/telescope tables for each muon event
+        """
+
+        # write the telescope tables
+
+        for tel_id, dl1_camera in event.dl1.tel.items():
+            dl1_camera.prefix = ""  # don't want a prefix for this container
+            telescope = self._subarray.tel[tel_id]
+            self.log.debug("WRITING TELESCOPE %s: %s", tel_id, telescope)
+
+            tel_index = TelEventIndexContainer(
+                obs_id=event.index.obs_id,
+                event_id=event.index.event_id,
+                tel_id=np.int16(tel_id),
+            )
+
+            if dl1_camera.muon_parameters is not None:
```

Comment:
this if should be first to not do operations like creating the container when it is not necessary.

You can avoid the additional indentation level by doing:

```
if dl1_camera.muon_parameters is None:
    continue
```
first